11_FFTBloom/main.cpp

```diff
@@ -0,0 +1,1192 @@
+			m_colMajorBuffer = m_device->createBuffer(std::move(deviceLocalBufferParams));
+
+			auto rowMemReqs = m_rowMajorBuffer->getMemoryReqs();
+			auto colMemReqs = m_colMajorBuffer->getMemoryReqs();
+			rowMemReqs.memoryTypeBits &= m_device->getPhysicalDevice()->getDeviceLocalMemoryTypeBits();
+			colMemReqs.memoryTypeBits &= m_device->getPhysicalDevice()->getDeviceLocalMemoryTypeBits();
+			auto gpuRowMem = m_device->allocate(rowMemReqs, m_rowMajorBuffer.get(), IDeviceMemoryAllocation::E_MEMORY_ALLOCATE_FLAGS::EMAF_DEVICE_ADDRESS_BIT);
+			auto gpuColMem = m_device->allocate(colMemReqs, m_colMajorBuffer.get(), IDeviceMemoryAllocation::E_MEMORY_ALLOCATE_FLAGS::EMAF_DEVICE_ADDRESS_BIT);
+
+			m_rowMajorBufferAddress = m_rowMajorBuffer.get()->getDeviceAddress();
+			m_colMajorBufferAddress = m_colMajorBuffer.get()->getDeviceAddress();
+		}
+
+		// Create pipeline layouts
+		auto createSampler = [&](ISampler::E_TEXTURE_CLAMP textureWrap) -> smart_refctd_ptr<IGPUSampler>
+		{
+			IGPUSampler::SParams params =
+			{
+				textureWrap,
+				textureWrap,
+				textureWrap,
+				ISampler::ETBC_FLOAT_OPAQUE_BLACK,
+				ISampler::ETF_LINEAR,
+				ISampler::ETF_LINEAR,
+				ISampler::ESMM_LINEAR,
+				3u,
+				0u,
+				ISampler::ECO_ALWAYS
+			};
+			return m_device->createSampler(std::move(params));
+		};
+
+		smart_refctd_ptr<IGPUPipelineLayout> imageFirstAxisFFTPipelineLayout;
+		{
+			auto sampler = createSampler(ISampler::E_TEXTURE_CLAMP::ETC_MIRROR);
+			IGPUDescriptorSetLayout::SBinding bnd =
+			{
+				IDescriptorSetLayoutBase::SBindingBase(),
+				0u,
+				IDescriptor::E_TYPE::ET_COMBINED_IMAGE_SAMPLER,
+				IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+				IShader::E_SHADER_STAGE::ESS_COMPUTE,
+				1u,
+				&sampler
+			};
+
+			imageFirstAxisFFTPipelineLayout = createPipelineLayout({ &bnd, 1 });
+		}
+
+		smart_refctd_ptr<IGPUPipelineLayout> lastAxisFFT_convolution_lastAxisIFFTPipelineLayout;
+		{
+			auto sampler = createSampler(ISampler::E_TEXTURE_CLAMP::ETC_MIRROR);
+			smart_refctd_ptr<IGPUSampler> samplers[CHANNELS];
+			std::fill_n(samplers, CHANNELS, sampler);
+			IGPUDescriptorSetLayout::SBinding bnd =
+			{
+				IDescriptorSetLayoutBase::SBindingBase(),
+				0u,
+				IDescriptor::E_TYPE::ET_COMBINED_IMAGE_SAMPLER,
+				IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+				IShader::E_SHADER_STAGE::ESS_COMPUTE,
+				CHANNELS,
+				samplers
+			};
+
+			lastAxisFFT_convolution_lastAxisIFFTPipelineLayout = createPipelineLayout({ &bnd, 1 });
+		}
+
+		smart_refctd_ptr<IGPUPipelineLayout> imageFirstAxisIFFTPipelineLayout;
+		{
+			IGPUDescriptorSetLayout::SBinding bnd =
+			{
+				IDescriptorSetLayoutBase::SBindingBase(),
+				0u,
+				IDescriptor::E_TYPE::ET_STORAGE_IMAGE,
+				IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+				IShader::E_SHADER_STAGE::ESS_COMPUTE,
+				1,
+				nullptr
+			};
+
+			imageFirstAxisIFFTPipelineLayout = createPipelineLayout({ &bnd, 1 });
+		}
```

Comment:
1. one pipeline layout
2. create the Descriptor Set using asset converter
3. create the sampler as CPUSampler and put it in the ICPUDescriptorSet before conversion

you can have a semi-unpopulated (unwritted bindings) in the ICPUDescriptorSet, and later write the storage images into the IGPUDescriptorSet resulting from the conversion